spring-integration-core/src/test/java/org/springframework/integration/json/SimpleJsonSerializerTests.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.json;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import java.util.Map;
+
+import org.junit.Test;
+
+import org.springframework.integration.support.json.JsonObjectMapperProvider;
+
+/**
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class SimpleJsonSerializerTests {
+
+	@Test
+	public void test() throws Exception {
+		Foo foo = new Foo();
+		String json = SimpleJsonSerializer.toJson(foo, "fileInfo");
+		@SuppressWarnings("unchecked")
+		Map<String, Object> fromJson = JsonObjectMapperProvider.newInstance().fromJson(json, Map.class);
```

Comment:
That's true, but if it is JSON String, the Jackson must be able to deserialize it into `Foo`.
I mean `.fromJson(json, Foo.class)`